assets/src/edit-story/components/library/panes/media/mediaPane.js

```diff
@@ -196,12 +200,20 @@ function MediaPane(props) {
   });
 
   /**
-   * Handle search term changes.
+   * Effectively performs a search, triggered at most every 500ms.
+   */
+  const [changeSearchTermDebounced] = useDebouncedCallback(() => {
```

Comment:
Why not just make `onSearch` debounced instead of introducing state?

Reply:
SearchInput is a controlled component (its displayed value is the "searchTerm" which triggers a search), so unless you decouple the current value displayed in the input with the value that is used to trigger the searches, it doesn't work in my testing.  If you debounce onSearch, what happens is that a keystroke takes 500ms to register and appear in the input, which is not what we want.